python-src/utility_objects.py

```diff
@@ -0,0 +1,95 @@
+from typing import Dict, List, Set
+import re
+import networkx as nx
+
+from pathlib import Path
+
+class IncludeDirectives:
+    ANGLE_PATTERN = re.compile(r'^\s*#\s*include\s*<(\w+)>\s*$')  # no spaces between angles and text?
+    QUOTE_PATTERN = re.compile(r'^\s*#\s*include\s*"(.+)"\s*$')  # .+ too naive?
+    def __init__(self) -> None:
+        self.angles: List[str] = []
+        self.quotes: List[str] = []
+
+    # return True iff `line` is an include directive.
+    def add_include_directive(self, line: str) -> bool:
+        angle_match = IncludeDirectives.ANGLE_PATTERN.match(line)
+        quote_match = IncludeDirectives.QUOTE_PATTERN.match(line)
+        if not (angle_match or quote_match):
+            return False
+        if angle_match:
+            self.angles.append(angle_match.group(1))
+        if quote_match:
+            self.quotes.append(quote_match.group(1))
+        return True
+
+
+class CppFile:
+    def __init__(self, absolute_path) -> None:
+        self.includes: IncludeDirectives = IncludeDirectives()
+        self.body_lines: List[str] = []
+        self.__path: Path = Path(absolute_path)  # throw if path not absolute?
```

Comment:
instead of using absolute path, we may give one reference path location, i.e. project location, together with the file location.